internal/util/typeutil/schema.go

```diff
@@ -395,6 +397,46 @@ func AppendFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx i
 	}
 }
 
+// DeleteFieldData delete fields data from src to dst
+func DeleteFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData, idx int64) {
+	for i, fieldData := range src {
+		switch fieldType := fieldData.Field.(type) {
+		case *schemapb.FieldData_Scalars:
+			dstScalar := dst[i].GetScalars()
+			switch fieldType.Scalars.Data.(type) {
+			case *schemapb.ScalarField_BoolData:
+				dstScalar.GetBoolData().Data = dstScalar.GetBoolData().Data[:len(dstScalar.GetBoolData().Data)-1]
```

Comment:
This could be panic if there's no data inside, please check the length and prevent panic from happening, just return if length==0.